experimental/ast/decl_file.go

```diff
@@ -70,6 +72,38 @@ func (f File) Imports() iter.Seq2[int, DeclImport] {
 	}
 }
 
+// ToProtoOptions contains options for the [File.ToProto] function.
+type ToProtoOptions struct {
+	// If set, no spans will be serialized.
+	//
+	// This operation only destroys non-semantic information.
+	ElideSpans bool
+
+	// If set, the contents of the file the AST was parsed from will not
+	// be serialized.
+	ElideFile bool
+}
+
+// ToProto converts this AST into a Protobuf representation, which may be
+// serialized.
+//
+// Note that package ast does not support deserialization from this proto;
+// instead, you will need to re-parse the text file included in the message.
+// This is because the AST is much richer than what is stored in this message;
+// the message only provides enough information for further semantic analysis
+// and diagnostic generation, but not for pretty-printing.
+//
+// Panics if the AST contains a cycle (e.g. a message that contains itself as
+// a nested message). Parsed ASTs will never contain cycles, but users may
+// modify them into a cyclic state.
+func (f File) ToProto(options *ToProtoOptions) proto.Message {
```

Comment:
Why a pointer for `ToProtoOptions`?

Reply:
So callers can pass in `nil` for all defaults. But I suppose that's an unnecessary vanity.